rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/statistics/ProbablisticSelectionManager.java

```diff
@@ -170,24 +170,27 @@ public ProbablisticSelectionManager(RMCore rmcore) {
      */
     @Override
     public synchronized boolean isPassed(SelectionScript script, Map<String, Serializable> bindings, RMNode rmnode) {
-        String digest;
+        String digest = null;
         SelectionScript scriptWithReplacedBindings = replaceBindings(script, bindings);
+        if (logger.isTraceEnabled()) {
```

Comment:
I thought "isTraceEnabled" should be used only if we going to spend a lot of resources to prepare log message.

Reply:
it displays the script contentq in that case, (id contains actually the script contents). String concatenation is linear time, but still.